.github/workflows/reusable-uki-test.yaml

```diff
@@ -121,38 +114,38 @@ jobs:
-          export DATASOURCE=${PWD}/build/datasource.iso
           export UPGRADE_IMAGE=$(cat "${PWD}/build/upgrade_image_name")
           export EXPECTED_NEW_VERSION=$(cat "${PWD}/build/expected_new_version")
           export EXPECTED_SINGLE_ENTRY="testentry"
           cp tests/go.* .
           go run github.com/onsi/ginkgo/v2/ginkgo -v --label-filter "uki" --fail-fast -r ./tests/
-      - name: Install kairos-agent (for versioneer)
-        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && inputs.flavor == 'ubuntu' }}
-        uses: Luet-lab/luet-install-action@cec77490c3f2416d7d07a47cfab04d448641d7ce # v1.1
-        with:
-          repository: quay.io/kairos/packages
-          packages: system/kairos-agent
       - name: Install Cosign
-        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && inputs.flavor == 'ubuntu' }}
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && steps.split.outputs.flavor == 'ubuntu' }}
         uses: sigstore/cosign-installer@main
       - name: Login to Quay Registry
-        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && inputs.flavor == 'ubuntu' }}
-        run: echo ${{ secrets.QUAY_PASSWORD }} | docker login -u ${{ secrets.QUAY_USERNAME }} --password-stdin quay.io
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && steps.split.outputs.flavor == 'ubuntu' }}
+        uses: docker/login-action@v3
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Set Image Tag
+        run: |
+          echo "GIT_VERSION=$(git describe --always --tags --dirty)" >> $GITHUB_ENV
+          echo "IMAGE_TAG=quay.io/kairos/${{ steps.split.outputs.flavor }}:${{ steps.split.outputs.flavor_release }}-${{ inputs.variant }}-${{ inputs.arch }}-${{ inputs.model }}-${{ env.GIT_VERSION }}${{ inputs.kubernetes_version && '-'||''}}${{ inputs.kubernetes_version || ''}}-uki" >> $GITHUB_ENV
+          echo "Setting image tag to ${{ env.IMAGE_TAG }}"
       - name: Push to quay
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && inputs.flavor == 'ubuntu' }}
         env:
           COSIGN_YES: true
         run: |
-          SUFFIX="-uki"
-          IMAGE=$(FLAVOR=${{ inputs.flavor }} FLAVOR_RELEASE="${{ inputs.flavor_release }}" MODEL=generic TARGETARCH=amd64 VARIANT=${{ inputs.variant }} REGISTRY_AND_ORG="quay.io/kairos" RELEASE=master kairos-agent versioneer container-artifact-name)
-          docker tag ${{ inputs.flavor }}-init:${{ inputs.flavor_release }} "$IMAGE$SUFFIX"
-          docker push "$IMAGE$SUFFIX"
-          image_ref=$(docker image inspect --format='{{index .RepoDigests 0}}' "$IMAGE$SUFFIX")
+          docker tag ${{ steps.split.outputs.flavor }}-init:${{ steps.split.outputs.flavor_release }} ${{ env.IMAGE_TAG }}
```

Comment:
Was it on purpose that this testing pipeline is pushing an image to `quay.io` ? All images to quay.io should come from the release pipeline I think (since they are, well, "released"). Unless we create some other org for temporary images like those we send to ttl.sh.

Reply:
`if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && inputs.flavor == 'ubuntu' }}`

This was not changed from the job, I think as a request from someone (Ben?) we started to push the ubuntu uki master images so they were more recent and he could consume them directly from there. I do agree that master images should NOT be pushed ever but....it is what it is.

Reply:
ok let's revisit this after merging.